### DISCUSSION TITLE
Question about SelectLabel types

### DISCUSSION THREAD

**Discussion by SCasarotto** [ariakit/ariakit#1279] [Questions] [answered]
So while converting a project to Ariakit I started to build out some reusable components (nearly all just base styling) and ran into what feels like a bug with the types on SelectLabel. I am not sure its a bug or if its just my lack of understanding. For that reason I figured a discussion was more appropriate.

So here is what I was attempting to do:

```tsx
import {
	SelectLabel as AriaSelectLabel,
	SelectLabelProps as AriaSelectLabelProps,
} from 'ariakit'

export const SelectLabel = (props: AriaSelectLabelProps) => {
	const { className = '', ...rest } = props
	return <AriaSelectLabel className={`mb-1 ${className}`} {...rest} />
}
```

When doing this, I get the following TS error:
<img width="1131" alt="image" src="https://user-images.githubusercontent.com/13336976/165179981-ce188b51-2d2f-4aad-9336-65e69d149805.png">

Looking into this further in `select-label.d.ts`:
```ts
export declare const useSelectLabel: import("ariakit-utils/types").Hook<SelectLabelOptions<"div">>;
export declare const SelectLabel: import("ariakit-utils/types").Component<SelectLabelOptions<"div">>;
export declare type SelectLabelOptions<T extends As = "div"> = Options<T> & {
    state: SelectState;
};
export declare type SelectLabelProps<T extends As = "label"> = Props<SelectLabelOptions<T>>;
```

This lead me to try:
```tsx
import {
	SelectLabel as AriaSelectLabel,
	SelectLabelProps as AriaSelectLabelProps,
} from 'ariakit'

export const SelectLabel = (props: AriaSelectLabelProps<'div'>) => {
	const { className = '', ...rest } = props
	return <AriaSelectLabel className={`mb-1 ${className}`} {...rest} />
}
```
which seemed to fix the TS error.

Finally this made me wonder if the types should be changed from:
```ts
export declare type SelectLabelProps<T extends As = "label"> = Props<SelectLabelOptions<T>>;
```
to 
```ts
export declare type SelectLabelProps<T extends As = "div"> = Props<SelectLabelOptions<T>>;
```

I'd love any thoughts or info about what I might not fully understand here.

**Comment by diegohaz** [comment, marked as answer]
Oh, it looks like a bug. Good catch. Created #1280.

**Comment by SCasarotto** [comment]
Oh awesome. I was getting worried I really didn't understand `as` and the way the typing works for it. 🙃  I'd be happy to make the PR for that if you would like. Thanks for the help

**Reply by SCasarotto** [reply]
PR created here https://github.com/ariakit/ariakit/pull/1281
